include/mips-emulator/instruction.hpp

```diff
@@ -141,7 +243,51 @@ namespace mips_emulator {
             jtype.address = address & MASK;
         }
 
+
+        // Opcode for coprocessor instructions
+        static constexpr uint8_t COP0 = 0b010000;
+        static constexpr uint8_t COP1 = 0b010001;
+        static constexpr uint8_t COP2 = 0b010010;
+
+        // No need for FPU register naming since it's just $f[number]
+        
+        // FPU R-Type
+        Instruction(const FPURTypeOp op, const uint8_t ft, const uint8_t fs, const uint8_t fd, const FPUFunc func) {
+            fpu_rtype.cop1 = COP1;
+            fpu_rtype.fmt = static_cast<uint8_t>(op);
```

Comment:
This parameter is larger than the field. Why is the parameter called op and the field called fmt?